pisces/stations.py

```diff
@@ -0,0 +1,291 @@
+    Site = tables.get("site", None) or Site
+    Sitechan = tables.get("sitechan", None) or Sitechan
+    Sensor = tables.get("sensor", None) or Sensor
+
+    # avoid nonsense inputs
+    if not any([Network, Affiliation]):
+        msg = "Network or Affiliation table required."
+        raise ValueError(msg)
+    
+    if any([netname, auth]) and not Network:
+        # Network keywords supplied, but no Network table present
+        # TODO: replace with pisces.exc.MissingTableError
+        msg = "Network table required."
+        raise ValueError(msg)
+
+    if any([sta, time_]) and not Affiliation:
+        # Affiliation keywords supplied, but no Affiliation table present
+        # TODO: replace with pisces.exc.MissingTableError
+        msg = "Affiliation table required."
+        raise ValueError(msg)
+    
+    # join Network and Affiliation if both present
+    if Network and Affiliation:
+        query = query.filter(Network.net == Affiliation.net)
+
+    # if query contains Site, Sitechan, or Sensor, append to Affiliation 
+    # in order of Site, else Sitechan, else Sensor
+    if Affiliation and Site:
+        query = query.filter(Site.sta == Affiliation.sta)
+    elif Affiliation and Sitechan:
+        query = query.filter(Sitechan.sta == Affiliation.sta)
+    elif Affiliation and Sensor:
+        query = query.filter(Sensor.sta == Affiliation.sta)
+
+    if net:
+        net = make_wildcard_list(net)
+        query = query.filter(or_(*[Network.net.like(nets) for nets in net]))
+    
+    if netname:
+        netname = make_wildcard_list(netname)
+        query = query.filter(or_(*[Network.netname.like(netnames) for netnames in netname]))
+    
+    if auth:
+        auth = make_wildcard_list(auth)
+        query = query.filter(or_(*[Network.net.like(auths) for auths in auth]))
+
+    if sta:
+        sta = make_wildcard_list(sta)
+        query = query.filter(or_(*[Affiliation.sta.like(stas) for stas in sta]))
+
+    if time_:
+        t1, t2 = time_
+        t1 = UTCDateTime(t1).timestamp if t1 else None
+        t2 = UTCDateTime(t2).timestamp if t2 else None
+        if t1:
+            query = query.filter(t1 <= Affiliation.endtime)
+        if t2:
+            query = query.filter(t2 >= Affiliation.time)
+
+    return query
+
+
+def filter_site(query, sta=None, chan=None, time_=None, staname = None, refsta = None, **tables):
+    """
+    Parameters
+    ----------
+    session : sqlalchemy.orm.session instance, bound
+    site : 
+    sitechan : 
+    stas : 
+    chans :
+    time_ :
+    endtime :  
+    with_query :
+    affiliation_name :
+    sensor_name:
+
+    Returns
+    -------
+    query
+        sdfs
+    
+    Notes:
+    ------
+    
+    """
+     # get desired tables from the query
+    Site, Sitechan, Sensor, Affiliation = _get_entities(query, "Site","Sitechan","Sensor","Affiliation")
+    # override if provided
+    # XXX: add a flag for the 'filter but don't include'
```

Comment:
I think that including a table via `**tables` instead of in the query (via `_get_entities`) implies the 'filter but don't include' functionality, so I an explicit flag is unnecessary.

Reply:
removed line